comixed-services/src/test/java/org/comixedproject/service/library/LibrarySelectionServiceTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * ComiXed - A digital comic book library management application.
+ * Copyright (C) 2022, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.service.library;
+
+import static junit.framework.TestCase.*;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.comixedproject.model.comicbooks.ComicBook;
+import org.comixedproject.service.comicbooks.ComicBookService;
+import org.comixedproject.service.comicbooks.ComicException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LibrarySelectionServiceTest {
+  private static final Long TEST_INCOMING_COMIC_ID = 717L;
+  private static final Long TEST_EXISTING_COMIC_ID = 129L;
+
+  @InjectMocks private LibrarySelectionService service;
+  @Mock private ComicBookService comicBookService;
+  @Mock private ComicBook comicBook;
+
+  private Set<Long> incomingIds = new HashSet<>();
+  private Set<Long> existingIds = new HashSet<>();
```

Comment:
Should instance variables be marked final? We don't do that elsewhere in the codebase.

Reply:
That's right. I've gotten into the habit now of setting all variables to final by default. So if I reassign my variable, the IDE alerts me and I either revise my code or remove the final if necessary. So it's an extra security and it allows to see also from the declaration that the variable will not be reassigned even locally. It also seems to me that it is a performance optimization too. But on this last point, I have just learned that it is not necessary to use final because there is since Java 8 the Effectively Final which allows the compiler to consider the variable as final if it is not reassigned.

I am not expert enough to decide what is better, so I validate the code as it is :) . To think in the future if it is better or not to explicitly declare final all the variables if they are not reassigned.